src/app/application.cpp

```diff
@@ -466,15 +466,23 @@ void Application::runExternalProgram(const BitTorrent::Torrent *torrent) const
     });
     proc.startDetached();
 #else // Q_OS_WIN
-    QStringList args = Utils::String::splitCommand(Preferences::instance()->getAutoRunProgram().trimmed());
+    const QString program = Preferences::instance()->getAutoRunProgram().trimmed();
+    QStringList args = Utils::String::splitCommand(program);
 
     if (args.isEmpty())
         return;
 
     for (QString &arg : args)
+    {
+        // strip redundant quotes
+        if (arg.startsWith(u'"') && arg.endsWith(u'"'))
```

Comment:
I think `Utils::String::splitCommand()` should return empty string for any empty parameter, either quoted or unquoted. Quotes aren't part of parameter, it is just a way to explicitly specify boundaries of some parameter(s) inside command line. Or am I wrong?

Reply:
>Quotes aren't part of parameter, it is just a way to explicitly specify boundaries of some parameter(s) inside command line

In general sense about command line you are right, but here it is not going through the command line. Instead it is going through `QProcess` blackbox and then to argc/argv. And it is quite confusing with the behavior of `QProcess`: on Windows when the quotes are stripped, the results are incomprehensible. On Linux the quotes should be stripped in order to produce command-line like behavior. So currently I'm inclined to think whether having quotes or not is platform dependent (or more like `QProcess` behavior dependent). And therefore I choose `Utils::String::splitCommand()` to not tangle with platform/QProcess specifics and keep its behavior simple.